### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by clickingbuttons**
Add github action to run `black` (linter) and `mypi` (static linter).

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: pr
```
Comment (morningvera) [inline .github/workflows/pr.yml:1]: repos generally break these up by workflow (i.e. run lint in a separate workflow from test and whatever else). this workflow name makes me think you're going to throw all of the steps into this file as things become more complex. is that correct? if not, i recommend renaming to `lint`

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+on: pull_request
```
Comment (morningvera) [inline .github/workflows/pr.yml:2]: this is the general pattern that i've seen in most repos

```suggestion
on:
  push:
    tags:
      - v*
    branches:
      - master
  pull_request:
```

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
```
Comment (morningvera) [inline .github/workflows/pr.yml:9]: 👍

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+name: pr
+on: pull_request
+jobs:
+  lint:
```
Comment (morningvera) [inline .github/workflows/pr.yml:4]: just fyi, i'm starting to see people set permissions on these. idk what kind of permissions these particular actions require but it's worth considering

```
permissions:
  contents: read
```

File: .github/workflows/pr.yml (hunk 1 above)
Comment (clickingbuttons) [reply .github/workflows/pr.yml:1]: I'm going to add testing in the next PR. What about the name "list-test"?

File: .github/workflows/pr.yml (hunk 2 above)
Comment (clickingbuttons) [reply .github/workflows/pr.yml:2]: Yep! I'll add releasing on the on push to master later.

File: .github/workflows/pr.yml (hunk 4 above)
Comment (clickingbuttons) [reply .github/workflows/pr.yml:10]: Will look at docs.

Comment (clickingbuttons) [reply .github/workflows/pr.yml]: Added.

**Review by morningvera — APPROVED** [review]

**Pull request merged**